src/main/java/com/cleanroommc/groovyscript/compat/mods/forestry/Centrifuge.java

```diff
@@ -0,0 +1,127 @@
+        return new RecipeBuilder();
+    }
+
+    @Override
+    @GroovyBlacklist
+    public void onReload() {
+        if (!isEnabled()) return;
+        removeScripted().forEach(CentrifugeRecipeManagerAccessor.getRecipes()::remove);
+        restoreFromBackup().forEach(CentrifugeRecipeManagerAccessor.getRecipes()::add);
+    }
+
+    public void add(ICentrifugeRecipe recipe) {
+        if (recipe == null) return;
+        addScripted(recipe);
+        CentrifugeRecipeManagerAccessor.getRecipes().add(recipe);
+    }
+
+    public boolean remove(ICentrifugeRecipe recipe) {
+        if (recipe == null) return false;
+        addBackup(recipe);
+        return CentrifugeRecipeManagerAccessor.getRecipes().remove(recipe);
+    }
+
+    public boolean removeByInput(IIngredient input) {
+        if (CentrifugeRecipeManagerAccessor.getRecipes().removeIf(recipe -> {
+            boolean found = input.test(recipe.getInput());
+            if (found) addBackup(recipe);
+            return found;
+        })) return true;
+
+        GroovyLog.msg("Error removing Forestry Centrifuge recipe")
+                .add("could not find recipe with input {}", input)
+                .error()
+                .post();
+        return false;
+    }
+
+    public boolean removeByOutputs(IIngredient... output) {
+        Set<ItemStack> list = Arrays.stream(output).map(i -> i.getMatchingStacks()[0]).collect(Collectors.toSet());
+        if (CentrifugeRecipeManagerAccessor.getRecipes().removeIf(recipe -> {
+            boolean found = list.containsAll(recipe.getAllProducts().keySet());
+            if (found) addBackup(recipe);
+            return found;
+        })) return true;
+
+        GroovyLog.msg("Error removing Forestry Centrifuge recipe")
+                .add("could not find recipe with outputs {}", list)
+                .error()
+                .post();
+        return false;
+    }
+
+    public boolean removeByOutput(IIngredient... output) {
+        return removeByOutputs(output);
+    }
+
+    public void removeAll() {
+        CentrifugeRecipeManagerAccessor.getRecipes().forEach(this::addBackup);
+        CentrifugeRecipeManagerAccessor.getRecipes().clear();
+    }
+
+    public SimpleObjectStream<ICentrifugeRecipe> streamRecipes() {
+        return new SimpleObjectStream<>(CentrifugeRecipeManagerAccessor.getRecipes()).setRemover(this::remove);
+    }
+
+    public class RecipeBuilder extends AbstractRecipeBuilder<ICentrifugeRecipe> {
+
+        protected int time = 20;
+        protected Map<ItemStack, Float> outputs = new Object2FloatOpenHashMap<>();
+
+        public RecipeBuilder time(int time) {
+            this.time = time;
+            return this;
+        }
+
+        public RecipeBuilder output(ItemStack output, float chance) {
+            this.outputs.put(output, chance);
+            return this;
+        }
+
+        @Override
+        public AbstractRecipeBuilder<ICentrifugeRecipe> output(ItemStack output) {
```

Comment:
why does this use `AbstractRecipeBuilder<ICentrifugeRecipe>` instead of `RecipeBuilder`?